src/analysis/retail/druid/restoration/modules/spells/Dreamstate.tsx

```diff
@@ -0,0 +1,61 @@
+import { TALENTS_DRUID } from 'common/TALENTS';
+import STATISTIC_ORDER from 'parser/ui/STATISTIC_ORDER';
+import STATISTIC_CATEGORY from 'parser/ui/STATISTIC_CATEGORY';
+import Analyzer, { Options, SELECTED_PLAYER } from 'parser/core/Analyzer';
+import SPELLS from 'common/SPELLS';
+import Events, { CastEvent } from 'parser/core/Events';
+import Statistic from 'parser/ui/Statistic';
+import { SpellIcon } from 'interface';
+import BoringValue from 'parser/ui/BoringValueText';
+import { formatNumber } from 'common/format';
+
+const cdrPerTick = 3;
+
+class Dreamstate extends Analyzer {
+  tickCount = 0;
+
+  constructor(options: Options) {
+    super(options);
+    this.active = this.selectedCombatant.hasTalent(TALENTS_DRUID.GROVE_TENDING_TALENT);
```

Comment:
```suggestion
    this.active = this.selectedCombatant.hasTalent(TALENTS_DRUID.GROVE_TENDING_TALENT);
    if (!this.active) { 
    		return;
    }
```

Reply:
I must fight the good fight! Doing this is optional! (I don't do this, and the listeners won't be added even if you don't return)

Reply:
Technically it prevents some function calls for slight performance boost (lol)

Reply:
But it's not pretty reeeeeee